### PR TITLE
zchunk: 1.1.9 -> 1.1.11

### PR DESCRIPTION
**Pull request by akino11111**
###### Motivation for this change
Update zchunk version

###### Things done
Updated zchunk version and hash from 1.1.9 to 1.1.11

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS **nix-build (Project version: 1.1.11)**
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) **Found none**
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`) **(zchunk 1.1.11 Copyright (c) 2021 Jonathan Dieter)**
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rb2k** [comment]
Result of `nixpkgs-review pr 126146` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>createrepo_c</li>
    <li>libdnf</li>
    <li>librepo</li>
    <li>microdnf</li>
    <li>python38Packages.librepo</li>
    <li>python39Packages.librepo</li>
    <li>zchunk</li>
  </ul>
</details>

**Comment by rb2k** [comment]
Result of `nixpkgs-review pr 126146` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>createrepo_c</li>
    <li>libdnf</li>
    <li>librepo</li>
    <li>microdnf</li>
    <li>python38Packages.librepo</li>
    <li>python39Packages.librepo</li>
    <li>rpm-ostree</li>
    <li>zchunk</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126146` at 8bce6746 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>librepo</li>
  </ul>
</details>
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>createrepo_c</li>
    <li>libdnf</li>
    <li>microdnf</li>
    <li>python38Packages.librepo</li>
    <li>python39Packages.librepo</li>
    <li>rpm-ostree</li>
    <li>zchunk</li>
  </ul>
</details>

File: pkgs/development/libraries/zchunk/default.nix [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
+  version = "1.1.11";
 
   outputs = [ "out" "lib" "dev" ];
 
   src = fetchFromGitHub {
     owner = "zchunk";
     repo = pname;
     rev = version;
-    hash = "sha256-MqnHtqOjLl6R5GZ4f2UX1iLoO9FUT2IfZlSN58wW8JA=";
+    sha256 = "1x7nrrhcafml6sgqr5h73v3hlcyq2275npngik4j6kr7wlj9dsmg";
```
Comment (AndersonTorres) [inline pkgs/development/libraries/zchunk/default.nix:22]:
```suggestion
    hash = "sha256-r+qWJOUnTyPJjM9eW44Q2DMKxx4HloyfNrQ6xWDO9vQ=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/zchunk/default.nix (hunk 1 above)
Comment (rb2k) [reply pkgs/development/libraries/zchunk/default.nix:22]: Hey @AndersonTorres!

I tried to understand the whole "when to use hash and when to use sha256" by looking at [the manual for fetchfromgithub](https://nixos.org/manual/nixpkgs/stable/#fetchfromgithub) it just mentions 'sha256' as an argument.

Why is hash preferred?

Comment (AndersonTorres) [reply pkgs/development/libraries/zchunk/default.nix:22]: For all intents and purposes, they are interchangeable. 

`hash` parameter allows other hash formats besides SHA256, therefore it is future-proof.
Nonetheless, there is no difference between using `sha256` and `hash` (except some minor annoyances).

On the other hand, some legacy fetchers, like `fetchgit, fetchsvn` &c, weren't updated yet. But they are rarely used, the chance you will come across one of them is small. Just use `hash` and be happy.

About the SRI format, `sha256-<crazy string>=`, it is just a new format for sharing hashes. The new `nixUnstable` tools use it by default.

Comment (akino11111) [reply pkgs/development/libraries/zchunk/default.nix]: Changed to hash

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Just squash the commits you made and force-push.

**Pull request merged**